### PR TITLE
nomacs: init at 3.4

### PR DESCRIPTION
**Pull request by ahmedtd**
###### Motivation for this change

Add nomacs-3.4, a Qt-based image viewer.

###### Things done

- [x] Tested using sandboxing
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/graphics/nomacs/default.nix [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  name = "nomacs-${version}";
+
+  enableParallelBuilding = true;
+
+  sourceRoot = "nomacs-3.4/ImageLounge";
+
+  patches = [./fix-appdata-install.patch];
+
+  buildInputs = [cmake
```
Comment (ndowens) [inline pkgs/applications/graphics/nomacs/default.nix]: Cmake pkgconfig and makeWrapper belong in nativeBuildInputs

**Review by ndowens — CHANGES_REQUESTED** [review]

File: pkgs/applications/graphics/nomacs/default.nix (hunk 1 above)
Comment (ahmedtd) [inline pkgs/applications/graphics/nomacs/default.nix]: OK, done.

File: pkgs/applications/graphics/nomacs/default.nix [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+, opencv
+, libraw
+, libtiff
+, quazip
+}:
+
+stdenv.mkDerivation rec {
+  version = "3.4";
+  src = fetchurl {
+    url = "https://github.com/nomacs/nomacs/archive/${version}.tar.gz";
```
Comment (Mic92) [inline pkgs/applications/graphics/nomacs/default.nix]: We prefer `fetchFromGitHub` usually here.

File: pkgs/applications/graphics/nomacs/default.nix [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+    sha256 = "552eda88aedea48831ce354095e3aad47892b4b5029f424171bedb68271c2a2f";
+    sha512 = "67a1b57971dc373d5a3be75b7deaff6702893252568eef135903754b2465416a58b40f18f55cf2994c8c3853ae96b82506c1caf26b0e645c20179a9cd81c0d36";
```
Comment (Mic92) [inline pkgs/applications/graphics/nomacs/default.nix]: There is no need for having both checksum. Only one checksum is evaluated at the moment.

**Pull request merged**